app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditPresenter.java

```diff
@@ -0,0 +1,345 @@
+                                id = id + set.getKey();
+                            }
+                        }
+
+                        if(list.containsKey(id)){
+
+                            //change prerequisites if they are affected
+                            getPrerequisites(new FirebaseCallback() {
+                                @Override
+                                public void onCallBack(HashMap<String, String> list) {
+                                    for(Map.Entry<String, String> set: list.entrySet()){
+
+                                        Log.i("EXECUTED", "NOWWWWWWWWWW");
+                                        Log.i(set.getKey(), set.getValue());
+
+                                        if(set.getValue().contains(id)){
+                                            Course course = new Course();
+                                            ArrayList arrayVersion = course.stringToArraylist(set.getValue());
+                                            for(int i = 0; i < arrayVersion.size(); i++){
+                                                if(arrayVersion.get(i).equals(id)){
+                                                    arrayVersion.remove(i);
+                                                }
+                                            }
+                                            String newPrerequisites = course.arraylistToString(arrayVersion);
+                                            offerings.child(set.getKey()).child("prerequisites").setValue(newPrerequisites);
+
+
+                                            offerings.child(id).removeValue();
+                                            warning.setTextColor(Color.GREEN);
+                                            warning.setText("Succesfully removed!");
+                                        }
+                                    }
+                                }
+                            });
+
+                        }
+                        else{
+                            warning.setText("Already removed!");
+                            warning.setTextColor(Color.RED);
+                        }
+                    }
+                });
+
+            }
+        });
+
+    }
+
+    //method to check is input is valid
+
+    public boolean isAllComma(String prerequisites){
+
+        int tracker = 0;
+        if(prerequisites.length() > 0) {
+            for (int i = 0; i < prerequisites.length(); i++) {
+                if (prerequisites.charAt(i) != ',') {
+                    tracker++;
+                }
+            }
+            if (tracker > 0) {
+                return false;
+            }
+            return true;
+        }
+        return false;
+    }
+
+    public boolean inputValidator(String courseCode, String courseName, boolean fall, boolean winter, boolean summer, String prerequisites){
+
+        //Initialize warning text reference
+        TextView warningText = view.getEditWarningText();
```

Comment:
if we had to conform to MVP for all elements of the app, the setting of text and view elements should live in the view, not the presenter. It should be fine for our purposes however

Reply:
I think the getEditWanrningText method has been defined in the view and is being used in the presenter to get the reference. Either way I don't think we need to strictly abide by it so we are good.

Reply:
I meant to refer to the rest of the code in the function as well, which set the text and colors. Those I believe belong in the view, but yeah nothing that neeeeds change